### PR TITLE
Add Bucket slice refactor to reduce re-renders

### PR DESCRIPTION
**Pull request by dvaldivia**
Signed-off-by: Daniel Valdivia <18384552+dvaldivia@users.noreply.github.com>

### CODE REVIEW

**Review by cniackz — DISMISSED** [review]
LGTM

File: portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketThunks.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          validity: retentionValidity,
+        };
+      }
+    }
+
+    return api
+      .invoke("POST", "/api/v1/buckets", request)
+      .then((res) => {
+        const newBucketName = `${bucketName}`;
+        dispatch(resetForm());
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketThunks.ts:77]: After bucket creation is not redirecting you to buckets list or object browser. it stays in the same create bucket page

File: portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketThunks.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        history.push(`/buckets/${newBucketName}/browse`);
+      })
+      .catch((err: ErrorResponseHandler) => {
+        dispatch(setErrorSnackMessage(err));
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketThunks.ts:81]: Invalid bucket names are not triggering any errors

**Review by bexsoft — CHANGES_REQUESTED** [review]

**Review by bexsoft — APPROVED** [review]
LGTM

**Review by bexsoft — APPROVED** [review]

**Review by adfost — APPROVED** [review]

**Pull request merged**